extensions/parcel/src/my-deliveries.tsx

```diff
@@ -46,57 +46,75 @@ export default function Command() {
     return `(in ${daysUntil} day${daysUntil !== 1 ? "s" : ""})`;
   };
 
-  // Format date in a more readable way: "Feb 26, 2025"
-  const formatFriendlyDate = (dateString: string): string => {
-    if (!dateString) return "Unknown date";
-
+  // Utility function to parse dates in various formats
+  const parseDate = (dateString: string): Date | null => {
     try {
-      const date = new Date(dateString);
+      let date: Date | null = null;
 
-      // Check if date is valid
-      if (isNaN(date.getTime())) {
-        return "Unknown date";
+      // First try American format (Month DD, YYYY HH:mm)
+      if (dateString.includes(",")) {
+        date = new Date(dateString);
       }
 
-      return date.toLocaleDateString(undefined, {
-        year: "numeric",
-        month: "short",
-        day: "numeric",
-      });
+      // If American format failed or wasn't used, try European format (DD.MM.YYYY HH:mm)
```

Comment:
Better to pull in https://date-fns.org if the dates aren't strictly formatted. It's much more readable to use formatting strings versus imperatively extracting each field from multiple formats.

```ts
import { parse, isValid } from 'date-fns';

export const parseDate = (str: string): Date | null => {
  for (const fmt of ['MMMM dd, yyyy HH:mm', 'dd.MM.yyyy HH:mm']) {
    const date = parse(str, fmt, new Date());
    if (isValid(date)) return date;
  }
  return null;
};
```

Reply:
Instead of `null` you may just want to return a string. But this is kind of a waste, if `parseDate` can't parse the date, it's fine to return an empty value. The decision to use the original string if the parsed version is empty is best left to the caller.